### PR TITLE
Angular springboot crud

### PR DESCRIPTION
**Pull request by jayeshsony**

### CODE REVIEW

File: Angular-Springboot-Crud-Backend/src/main/java/com/AngularSpringbootCrudApplication.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class AngularSpringbootCrudApplication {
+
+	/**
+	 * Main method to start the Spring Boot application.
+	 * 
+	 * @param args Command line arguments passed to the application.
+	 */
+	public static void main(String[] args) {
+		SpringApplication.run(AngularSpringbootCrudApplication.class, args);
+
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/java/com/AngularSpringbootCrudApplication.java]: Remove this line

File: Angular-Springboot-Crud-Backend/src/main/java/com/controller/EmployeeController.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	@ExceptionHandler(MethodArgumentNotValidException.class)
+	public ResponseEntity<Map<String, String>> handleValidationError(MethodArgumentNotValidException currentException) {
+		Map<String, String> errorsMap = new HashMap<>();
+
+		currentException.getBindingResult().getFieldErrors().forEach(error -> {
+			String field = error.getField();
+			String message = error.getDefaultMessage();
+			errorsMap.putIfAbsent(field, message);
+		});
+
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/java/com/controller/EmployeeController.java]: Remove this line

File: Angular-Springboot-Crud-Backend/src/main/java/com/dto/constantmessages/ConstantMessages.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+	public static final String GENDER_REQUIRED = "Gender is required";
+	public static final String GENDER_SIZE = "Gender must be between 4 and 10 characters";
+
+	public static final String MOBILE_NUMBER_REQUIRED = "Mobile number is required";
+	public static final String MOBILE_NUMBER_PATTERN = "Mobile number must be 10 digits";
+
+	public static final String USER_NOT_FOUND = "User not found";
+
+	private ConstantMessages() {
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/java/com/dto/constantmessages/ConstantMessages.java]: No need to add this

File: Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class EmployeeServiceImpl implements EmployeeService {
+
+	EmployeeRepository empRepo;
+	ModelMapper modelMapper;
+
+	@Override
+	public ResponseEntity<?> createOrUpdateEmployee(EmployeeDto employeeDTO) {
+		Long id = employeeDTO.getId();
+		if (id != null && empRepo.existsById(id)) {
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java]: Add proper name

File: Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+ * 
+ * @author Jayesh Soni
+ * @since 2024-06-04
+ */
+@Service
+@RequiredArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class EmployeeServiceImpl implements EmployeeService {
+
+	EmployeeRepository empRepo;
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java:39]: Add blank line after this

File: Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+	ModelMapper modelMapper;
+
+	@Override
+	public ResponseEntity<?> createOrUpdateEmployee(EmployeeDto employeeDTO) {
+		Long id = employeeDTO.getId();
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java]: You can write below like  employeeDTO.getId() != null

File: Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+			} else if (empRepo.existsByEmail(employeeDTO.getEmail())) {
+				return ResponseEntity.ok("emailExists");
+			} else if (empRepo.existsByMobileNumber(employeeDTO.getMobileNumber())) {
+				return ResponseEntity.ok("mobileExists");
+			}
+		}
+
+		Employee employee = modelMapper.map(employeeDTO, Employee.class);
+		Employee savedEmployee = empRepo.save(employee);
+		return ResponseEntity.ok(modelMapper.map(savedEmployee, EmployeeDto.class));
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java:67]: You can write  ResponseEntity.ok(modelMapper.map(empRepo.save(employee), EmployeeDto.class));

File: Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+	}
+
+	@Override
+	public List<Employee> retriveAllEmployees() {
+		List<Employee> allEmployees = empRepo.findAll(Sort.by(Direction.DESC, "id"));
+		return allEmployees;
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java]: You can return directly  empRepo.findAll(Sort.by(Direction.DESC, "id"));

File: Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+			// if id is not present throw error
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, ConstantMessages.USER_NOT_FOUND);
+		}
+	}
+
+	@Override
+	public Employee getEmployeeById(Long id) {
+		// Check if the id is present in database or not
+		if (empRepo.existsById(id)) {
+			Optional<Employee> foundEmployee = empRepo.findById(id);
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/java/com/service/impl/EmployeeServiceImpl.java]: Please add proper name

File: Angular-Springboot-Crud-Backend/src/main/resources/application.properties [hunk 10]
```diff
@@ -0,0 +1,13 @@
+spring.application.name=Angular-Springboot-Crud
+spring.mvc.view.prefix=/views/
+spring.mvc.view.suffix=.jsp
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Backend/src/main/resources/application.properties]: Add blank line after this and removed unnecessary properties

File: Angular-Springboot-Crud-Frontend/src/app/employee-list/employee-list.component.html [hunk 11]
```diff
@@ -0,0 +1,68 @@
+<div class="container-fluid">
+  <div class="nav-bar">
+    <div class="row">
+      <div class="col-3">
+        <ul style="display: flex; justify-content: space-around; list-style: none; ">
+          <li><a style="text-decoration: none;" routerLink="../create" routerLinkActive="active"><svg
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Frontend/src/app/employee-list/employee-list.component.html]: Removed all incline css

File: Angular-Springboot-Crud-Frontend/src/app/employee.ts [hunk 12]
```diff
@@ -0,0 +1,33 @@
+export class Employee {
+
+    // Unique identifier for the employee
+    id: number = 0;
+
+    // Employee's first name
+    firstName: string = "";
```
Comment (KrushaliTalaviya) [inline Angular-Springboot-Crud-Frontend/src/app/employee.ts]: You can write like below firstName!: string;

**Review by KrushaliTalaviya — CHANGES_REQUESTED** [review]

**Comment by jayeshsony** [comment]
Created PR cahnges

**Comment by jayeshsony** [comment]
Created PR changes

**Review by KrushaliTalaviya — APPROVED** [review]

**Pull request merged**